lib/fs/fs.c

```diff
@@ -159,12 +174,152 @@ result_t trn_fs_realpath(const char *path, char **resolved_path) {
 	return r;
 }
 
+void get_path_parent_child(char *path, char **out_parent, size_t *out_len_parent, char **out_child, size_t *out_child_len) {
+	char *path_end = path + strlen(path) - 1;
+	char *child;
+
+	while (path < path_end && *path_end == '/') {
+		// Skip trailing slashes
+		path_end--;
+	}
+
+	child = path_end;
+	while (path < child && *child != '/') {
+		// Get child path
+		child--;
+	}
+	*out_child = child + 1;
+	*out_child_len = (path_end - child + 1);
+	if (path < child - 1) {
+		// We have a parent
+		*out_parent = path;
+		*out_len_parent = child - path;
+	} else {
+		// Parent is CWD
+		*out_parent = "";
+		*out_len_parent = 0;
+	}
+}
+
+result_t trn_fs_mkdir(const char *path) {
+	result_t r;
+	trn_traverse_t traverse[MAX_RECURSION];
+	int borrowed_recursion = 0;
+	int traverse_recursion = 0;
+	char *parent_path, *child_path;
+	size_t parent_path_size, child_path_size;
+
+	get_path_parent_child(path, &parent_path, &parent_path_size, &child_path, &child_path_size);
+	printf("Parent is %.*s, child is %.*s\n", parent_path_size, parent_path, child_path_size, child_path);
```

Comment:
remove or put behind a `TRN_FS_DEBUG` macro, and verify that `child_path_size` > 0